recirq/kpz/experiment.py

```diff
@@ -0,0 +1,538 @@
+        ax.set_xlabel("Number of 1s that crossed center, $\mathcal{M}/2$")
+        ax.set_ylabel("Probability")
+        return ax
+
+
+class KPZExperimentResults:
+    """A class for processing and storing KPZ experiment results.
+
+    An object of this type is returned by `KPZExperiment.run_experiment()`, which uses
+    the `run()` method of the Cirq sampler to sample final bitstrings given the initial
+    bitstrings specified by the initial_states array. The outputs of `sampler.run()` are
+    inputted as `raw_results`. Pooling these results together, the transferred
+    magnetization is computed. Its histogram can be visualized using the `plot_histogram()`
+    method. The first four moments can also be computed, as well as their statistical
+    uncertainties.
+    """
+
+    def __init__(
+        self,
+        raw_results: List[List[cirq.study.result.ResultDict]],
+        initial_states: np.ndarray,
+    ):
+        num_trials, n = initial_states.shape
+        num_right = np.sum(initial_states[:, n // 2 :], 1)
+        self.num_initial_states = num_trials
+        self.num_right_initial = num_right
+
+        bitstrs = np.array([res.measurements["m"] for res in raw_results])
+        self.num_right_final = np.sum(bitstrs[:, :, n // 2 :], 2)
+
+        self.transferred_magnetization = self._transferred_magnetization()
+        self.mean = self._mean()
+        self.variance = self._variance()
+        self.skewness = self._skewness()
+        self.kurtosis = self._kurtosis()
+
+    def _transferred_magnetization(self) -> np.ndarray:
+        final = self.num_right_final
+        num_trials, num_reps = final.shape
+        initial = np.outer(self.num_right_initial, np.ones(num_reps, dtype=int))
+        return 2 * (final - initial)
+
+    def _mean(self) -> float:
+        return np.mean(self.transferred_magnetization.flatten())
+
+    def _variance(self) -> float:
+        return np.var(self.transferred_magnetization.flatten())
+
+    def _skewness(self) -> float:
+        return sstats.skew(self.transferred_magnetization.flatten())
+
+    def _kurtosis(self) -> float:
+        return sstats.kurtosis(self.transferred_magnetization.flatten(), fisher=True)
+
+    def _mean_excluding_i(self, i: int, axis: Optional[int] = 0) -> float:
+        tm = np.delete(self.transferred_magnetization, i, axis=axis)
+        return np.mean(tm.flatten())
+
+    def _variance_excluding_i(self, i: int, axis: Optional[int] = 0) -> float:
+        tm = np.delete(self.transferred_magnetization, i, axis=axis)
+        return np.var(tm.flatten())
+
+    def _skew_excluding_i(self, i: int, axis: Optional[int] = 0) -> float:
+        tm = np.delete(self.transferred_magnetization, i, axis=axis)
+        return sstats.skew(tm.flatten())
+
+    def _kurtosis_excluding_i(self, i: int, axis: Optional[int] = 0) -> float:
+        tm = np.delete(self.transferred_magnetization, i, axis=axis)
+        return sstats.kurtosis(tm.flatten(), fisher=True)
```

Comment:
Optional:  you could probably combine these into one function, i.e.

```
    def _stats_excluding_i(self, stats_function: Callable[np.array, float], i: int, axis: Optional[int] = 0) -> float:
        tm = np.delete(self.transferred_magnetization, i, axis=axis)
        return stats_function(tm.flatten())
```

I'm not sure if that would be more confusing or more readable though, so I'll leave it up to you.

Reply:
I considered that, but I do think it's clearer as is, so I will opt to leave it.